libr/bin/format/elf/elf.c

```diff
@@ -293,14 +359,21 @@ static int init_dynamic_section (struct Elf_(r_bin_elf_obj_t) *bin) {
 		return false;
 	if (bin->phdr[i].p_offset + sizeof(Elf_(Dyn)) > bin->size)
 		return false;
-	tmp = dyn = (Elf_(Dyn)*)((ut8 *)bin->b->buf + bin->phdr[i].p_offset);
-	for (entries = 0; (ut8*)dyn < ((ut8*)tmp + dyn_size); dyn++) {
-	    	entries++;
-		if (dyn->d_tag == DT_NULL) {
+
+	for (entries = 0; entries < (dyn_size / sizeof (Elf_(Dyn))); entries++) {
+		len = r_buf_read_at (bin->b, bin->phdr[i].p_offset +
+				entries * sizeof (Elf_(Dyn)), sdyn, sizeof (Elf_(Dyn)));
+		if (len < 1)
+			goto beach;
+		j = 0;
+#if R_BIN_ELF64
+		d.d_tag = READ64(sdyn, j)
```

Comment:
what about add a new macro for READ32 or READ64 depending on R_BIN_ELF64 ? and avoid that dup thing